tests/sdk.spec.ts

```diff
@@ -70,64 +70,128 @@ describe('HuddlySDK', () => {
   });
 
   describe('#setupDeviceDiscoveryListeners', () => {
+    const dummySerial = '123456';
     const dummyGO = {
-      productId: 0x11
+      productId: 0x11,
+      serialNumber: dummySerial
     };
     const dummyIQ = {
-      productId: 0x21
+      productId: 0x21,
+      serialNumber: dummySerial
     };
     let huddlygoInitStub;
     let boxfishInitStub;
     let discoveryEmitter;
     let otherEmitter;
     let hidInterfaceStub;
+    let dummyTargetSerial;
+
+    function initSdk() {
+      new HuddlySdk(nodeusbDeviceApi, [nodeusbDeviceApi], { emitter: otherEmitter, apiDiscoveryEmitter: discoveryEmitter, serial: dummyTargetSerial });
+    }
+
     beforeEach(() => {
       huddlygoInitStub = sinon.stub(HuddlyGo.prototype, 'initialize').resolves();
       boxfishInitStub = sinon.stub(Boxfish.prototype, 'initialize').resolves();
       hidInterfaceStub = sinon.stub(DeviceFactory, 'getHIDInterface').resolves();
       discoveryEmitter = new EventEmitter();
       otherEmitter = new EventEmitter();
-      new HuddlySdk(nodeusbDeviceApi, [nodeusbDeviceApi], { emitter: otherEmitter, apiDiscoveryEmitter: discoveryEmitter });
+      dummyTargetSerial = undefined;
     });
     afterEach(() => {
       huddlygoInitStub.restore();
       boxfishInitStub.restore();
       hidInterfaceStub.restore();
+
     });
 
     it('should emit ATTACH event with Boxfish instance when attached device is boxfish', (done) => {
+      initSdk();
+      otherEmitter.on('ATTACH', (d) => {
+        expect(d).to.be.instanceof(Boxfish);
+        done();
+      });
+      discoveryEmitter.emit('ATTACH', dummyIQ);
+    });
+    it('should emit ATTACH event with Boxfish instance when attached device is boxfish and targetSerial match', (done) => {
+      dummyTargetSerial = dummySerial;
+      initSdk();
       otherEmitter.on('ATTACH', (d) => {
         expect(d).to.be.instanceof(Boxfish);
         done();
       });
       discoveryEmitter.emit('ATTACH', dummyIQ);
     });
+
+    it('should not emit ATTACH event when device api emits attach with serialNumber not matching targetSerial', (done) => {
+      dummyTargetSerial = 'nonMatchSerial';
+      initSdk();
+      const attachSpy = sinon.spy();
+      otherEmitter.on('ATTACH', attachSpy);
+      discoveryEmitter.emit('ATTACH', dummyIQ);
+
+      setTimeout(() => {
+        expect(attachSpy.callCount).to.equals(0);
+        done();
+      }
+      , 50);
+    });
+
+    it('should emit ATTACH event with empty targetSerial', (done) => {
+      dummyTargetSerial = '';
+      initSdk();
+      otherEmitter.on('ATTACH', (d) => {
+        expect(d).to.be.instanceof(Boxfish);
```

Comment:
assert that the serial number is empty: `expect(d.serialNumber).to.equals(dummyTargetSerial);`

Reply:
This test was maybe a bit misleading. I wanted to clarify what happened if you gave an empty string as the targetSerial. In this case the SDK will just pick up any device regardless of serial number.